### PR TITLE
Adding Travis-Ci Support For Arm64

### PR DESCRIPTION
**Pull request by ossdev07**
Added support for ARM64 in Travis-CI.
Changes done in setup_conda_linux.sh and setup_dependencies_common.sh files.

### CODE REVIEW

**Comment by ossdev07** [comment]
Ping @bsipocz kindly have a look on this PR.

**Comment by ossdev07** [comment]
Ping @bsipocz kindly have a look on this commit

**Comment by ossdev07** [comment]
hi @bsipocz ,
Can you please have a look at above PR.

**Comment by bsipocz** [comment]
Hi @ossdev07!

thanks for the PR. Can I ask what package do you plan to support with these additions? Is the CI there relies on conda, or pip, or tox, or something else?

 Before going into the nitty gritty of reviewing it (unfortunately it adds some complexity to the over complicated script, including non negligible repetitions) , it would be nice to see the bigger picture.

**Comment by ossdev07** [comment]
Hi @bsipocz, 

Thanks for the reply.

These additions are required for below packages :
i) pytest-arraydiff
ii) pytest remotedata
iii) astropy 

These 3 packages are using ci-helpers package for the conda environment setup and dependencies installation in Travis. 
There are some repetitions of permission as these are required by the archiconda package for its successful execution.

**Comment by ossdev07** [comment]
Ping @bsipocz
 kindly have a look on this commit

**Comment by bsipocz** [comment]
@ossdev07 - thanks again for the PR.  Unfortunately, as I tried to say above, this doesn't really follow the logic we tried to do before, rather adds extra complexity. I don't have enough bandwith atm to do a thorough review, or reimplementation. Given you say this is motivated by the needs of astropy related packages, it definitely puts the feature to the backburner for me, as I know we are not bottlenecked in astropy by ci-helpers lack of arm support.

**Comment by ossdev07** [comment]
Hi @bsipocz

I have added another commit for replacing **archiconda** with **miniforge** which leads to the **removal of all the permission issues** now and the code has the **minimal changes** with the master branch. Kindly go through our commit and let me know if any further changes are required.

**Comment by ossdev07** [comment]
Hi @bsipocz,

I have rebased and removed redundant code. Please review and share your opinion.

**Comment by ossdev07** [comment]
@bsipocz kindly have a look on this commit and let us know if any changes required.

**Comment by odidev** [comment]
@bsipocz could you please have a look at this PR and let me know if any changes required.

File: travis/setup_conda_linux.sh [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
+   # See https://github.com/ContinuumIO/anaconda-issues/issues/11148
+fi
 mkdir $HOME/.conda
 bash miniconda.sh -b -p $HOME/miniconda
 $HOME/miniconda/bin/conda init bash
 source ~/.bash_profile
-conda activate base
-
+export PATH=$HOME/miniconda/bin/:$PATH
+source activate base
```
Comment (astrofrog) [inline travis/setup_conda_linux.sh:21]: Why is this change needed? I thought conda activate was The New Way :tm:?

File: travis/setup_conda_linux.sh [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
 # Install common Python dependencies
 source "$( dirname "${BASH_SOURCE[0]}" )"/setup_dependencies_common.sh
-
```
Comment (astrofrog) [inline travis/setup_conda_linux.sh:20]: Revert this change

File: travis/setup_dependencies_common.sh [hunk 3]
```diff
@@ -345,10 +349,9 @@ fi
 # Pin required versions for dependencies, howto is in FAQ of conda
 # https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-pkgs.html#preventing-packages-from-updating-pinning
 if [[ ! -z $CONDA_DEPENDENCIES ]]; then
-
-    if [[ -z $(echo $CONDA_DEPENDENCIES | grep '\bmkl\b') &&
-            $TRAVIS_OS_NAME != windows && ! -z $NUMPY_VERSION ]]; then
-        CONDA_DEPENDENCIES=${CONDA_DEPENDENCIES}" nomkl"
+        if [[ -z $(echo $CONDA_DEPENDENCIES | grep '\bmkl\b') &&
+                    $TRAVIS_OS_NAME != windows && ! -z $NUMPY_VERSION ]]; then
+                CONDA_DEPENDENCIES=${CONDA_DEPENDENCIES}" nomkl"
```
Comment (astrofrog) [inline travis/setup_dependencies_common.sh:354]: Revert since just a whitespace fix

File: travis/setup_dependencies_common.sh [hunk 4]
```diff
@@ -477,8 +480,11 @@ fi
 
 # We use --no-pin to avoid installing other dependencies just yet.
 
-
-MKL='nomkl'
+if [ `uname -m` = aarch64]; then
```
Comment (astrofrog) [inline travis/setup_dependencies_common.sh:483]: Add this as an or (||) clause below instead of having a separate if statement

Comment (astrofrog) [reply travis/setup_dependencies_common.sh:483]: Also use ``TRAVIS_CPU_ARCH`` instead of uname,

File: travis/setup_dependencies_common.sh [hunk 5]
```diff
@@ -190,7 +190,11 @@ fi
 # We pin the version for conda as it's not the most stable package from
 # release to release. Add note here if version is pinned due to a bug upstream.
 if [[ -z $CONDA_VERSION ]]; then
-    CONDA_VERSION=4.7.11
+    if [ `uname -m` = 'aarch64' ]; then
```
Comment (astrofrog) [inline travis/setup_dependencies_common.sh:193]: Use TRAVIS_CPU_ARCH

File: travis/setup_conda_linux.sh [hunk 6]
```diff
@@ -1,23 +1,26 @@
...
-# Create .conda directory before install to workaround conda bug
-# See https://github.com/ContinuumIO/anaconda-issues/issues/11148
+if [ `uname -m` = 'aarch64' ]; then
+   wget -q "https://github.com/conda-forge/miniforge/releases/download/4.8.2-1/Miniforge3-4.8.2-1-Linux-aarch64.sh" -O miniconda.sh
+   chmod +x miniconda.sh
+else
+   wget https://repo.continuum.io/miniconda/Miniconda3-${MINICONDA_VERSION}-Linux-x86_64.sh -O miniconda.sh --progress=dot:mega
+   # Create .conda directory before install to workaround conda bug
+   # See https://github.com/ContinuumIO/anaconda-issues/issues/11148
+fi
```
[the remark this answers is not in the document]
Comment (astrofrog) [reply travis/setup_conda_linux.sh:15]: Also use TRAVIS_CPU_ARCH

**Review by astrofrog — CHANGES_REQUESTED** [review]
Thanks for simplifying this! I think this looks a lot more manageable and some of the changes can be simplified further. Can you also rebase?

**Comment by pllim** [comment]
Due to recent developments, we have decided not to accept this. Sorry for any inconvenience caused!

Please see https://groups.google.com/g/astropy-dev/c/45frnTEAX-U and astropy/astropy#11045

**Pull request closed without merging**